### PR TITLE
[PR1] DEV-163 Added Optimized ReceiveTable

### PR DESCRIPTION
**Pull request by smorsua**
This is the definitive version of the ReceiveTable, which imperatively mutates the UI. It was inspired by an article that explains [the fastest way of mutating a text node](https://electricui.com/blog/fast-react-text-updates).

We first build the table using the data from the redux store and then update it in intervals. We take advantage of the fact that the boards in the redux store will never change. If in the future they do change, the component can be adapted to support this behaviour while mantaining the same performance.

https://user-images.githubusercontent.com/114338434/236465404-dce20cde-ee41-481e-94a6-e717815f6731.mp4

### CODE REVIEW

**Review by jmaralo — APPROVED** [review]

File: src/components/ReceiveTableContainer/ReceiveTable/BoardView/PacketView/MeasurementView/MeasurementView.tsx [hunk 1]
```diff
@@ -1,45 +1,46 @@
...
+        isNumeric
+            ? measurement.value.average.toFixed(3)
+            : measurement.value.toString()
+    );
+
     return (
-        // <div className={styles.measurementView}>
         <>
             <span className={styles.name}>{measurement.name}</span>
             {isNumeric && (
```
Comment (Losina24) [inline src/components/ReceiveTableContainer/ReceiveTable/BoardView/PacketView/MeasurementView/MeasurementView.tsx:25]: u can move this into a new component

File: src/components/ReceiveTableContainer/ReceiveTable/Header/Header.tsx [hunk 2]
```diff
@@ -46,11 +47,12 @@ export const Header = ({ items }: Props) => {
                             <Separator
                                 onMouseDown={(ev) => handleMouseDown(index, ev)}
                             ></Separator>
-                        </>
+                        </Fragment>
                     );
                 } else {
```
Comment (Losina24) [inline src/components/ReceiveTableContainer/ReceiveTable/Header/Header.tsx]: this 'else' is unnecessary. A more readable solution:

```ts
...
if (index < items.length - 1) return componentX(index)
return componentY(index)
}

const componentX = (index: Number) => ( // Change componentX to a more descriptive name
    <>
    <Fragment key={index}>
        <div 
                 className={styles.cell}
                 style={{ flexBasis: columns[index] }}
         >
             {item}
         </div>
         <SeparatoronMouseDown={(ev) => handleMouseDown(index, ev)}></Separator>
         </>
    </Fragment>
)

const componentY = (index: Number) => (
...
)
```

File: src/components/ReceiveTableContainer/ReceiveTable/ReceiveTable.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+type Props = {
+    boards: Record<string, Board>;
+};
+
+export const ReceiveTable = ({ boards }: Props) => {
+    return (
+        <TableUpdater>
+            <div className={styles.newReceiveTable}>
+                <Header items={["ID", "NAME", "COUNT", "CYCLE (ns)"]} />
```
Comment (Losina24) [inline src/components/ReceiveTableContainer/ReceiveTable/ReceiveTable.tsx:17]: Maybe its better to define the array in a const outside the html and pass the const here

File: src/components/ReceiveTableContainer/ReceiveTable/TableUpdater.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        }
+
+        for (const id in measurementElements.current) {
+            const measurement = state.measurements[id];
+            measurementElements.current[id].value.nodeValue =
+                isNumericMeasurement(measurement)
+                    ? measurement.value.average.toFixed(3)
+                    : measurement.value.toString();
+        }
+    }, 1000 / 60);
```
Comment (Losina24) [inline src/components/ReceiveTableContainer/ReceiveTable/TableUpdater.tsx]: Magic number. It is better to define this outside as a constant with a name that explains what that value is (const TICK_RATE = 1000/60 or whatever)

File: src/components/ReceiveTableContainer/ReceiveTable/BoardView/PacketView/MeasurementView/MeasurementView.tsx (hunk 1 above)
Comment (smorsua) [reply src/components/ReceiveTableContainer/ReceiveTable/BoardView/PacketView/MeasurementView/MeasurementView.tsx:25]: In this case, I think its it's clearer in this format.

**Review by lipezaballa — APPROVED** [review]

**Pull request merged**